### PR TITLE
Possibility to bake mesh during runtime with multiple grids

### PR DESCRIPTION
**Pull request by codxr02**
Hi h8man or vhman,

I adjusted the solution of DiegoMartinezCela [https://github.com/h8man/NavMeshPlus/issues/29](url) to allow multiple grids to be baked. Therefore, I added the **NavMeshGrid** component which kinda subscribes and unsubscribes itself to/from the NavMeshBuilder2d during runtime. The NavMeshBuilder2d now stores a **list of registeredGrids**, which it passes to NavMeshBuilder2dWrapper.GetRoot(), who'll simply return the registeredGrids. Even if no NavMeshGrid is used inside the scene (the length of registeredGrids will be equal to 0), the old way of finding 1 Grid via FindObjectOfTypey<Grid>() applies. The NavMeshBuilder2d now exposes a function called ClearAllRegisteredGrids(), which can be used to reset all registeredGrids during runtime.

How to use NavMeshGrid: All you have to do is adding the NavMeshGrid component to all the Grids available in your scene (or will be Instantiated at some point during runtime). To bake the grid during runtime you can use the following code:
```
if (surface.navMeshData != null) {
    surface.UpdateNavMesh(surface.navMeshData);
}
else {
    surface.BuildNavMesh();
}
```
I thought I share my solution with everyone since it's working like a charm for me.

### CODE REVIEW

**Comment by 2shady4u** [comment]
Any update on merging this?

**Comment by codxr02** [comment]
@2shady4u Weird, the build works totally fine on my end... I removed the #if UNITY EDITOR blocks now though. :)

File: NavMeshComponents/Scripts/NavMeshBuilder2d.cs [hunk 1]
```diff
@@ -19,6 +19,7 @@ class NavMeshBuilder2dWrapper
         public CollectObjects2d CollectObjects;
         public GameObject parent;
         public bool hideEditorLogs;
+        public bool hideEditorLogs;
```
Comment (tom-sosedow) [inline NavMeshComponents/Scripts/NavMeshBuilder2d.cs:22]: why twice?

**Comment by swampkid** [comment]
The Compile Error I got, was caused by a double variable declaration(hideEditorLogs) in NavMeshBuilder2d on line 22.
Delete this and it works like a charm!

**Comment by h8man** [comment]
https://github.com/h8man/NavMeshPlus/issues/29 
https://github.com/h8man/NavMeshPlus/pull/73

**Pull request closed without merging**